src/status_im/ui/screens/signing/views.cljs

```diff
@@ -0,0 +1,197 @@
+        (if in-progress?
+          [react/activity-indicator {:animating true
+                                     :size      :large}]
+          [button/primary-button {:on-press  #(re-frame/dispatch [:signing.ui/sign-is-pressed])
+                                  :disabled? (not enabled?)}
+           (i18n/label :t/transactions-sign)])]]
+      [react/view])))
+
+(views/defview message-sheet []
+  (views/letsubs [{:keys [formatted-data] :as sign} [:signing/sign]]
+    [react/view styles/message
+     [react/view styles/message-header
+      [react/text {:style {:typography :title-bold}} (i18n/label :t/signing-a-message)]
+      [react/touchable-highlight {:on-press #(re-frame/dispatch [:signing.ui/cancel-is-pressed])}
+       [react/text {:style {:color colors/blue}} (i18n/label :t/cancel)]]]
+     [separator]
+     [react/view {:padding-top 16}
+      [react/view styles/message-border
+       [react/scroll-view
+        [react/text (or formatted-data "")]]]
+      [password-view sign]]]))
+
+(views/defview sheet [{:keys [contact amount token] :as tx}]
+  (views/letsubs [fee   [:signing/fee]
+                  sign  [:signing/sign]
+                  chain [:ethereum/chain-keyword]
+                  {:keys [amount-error gas-error]} [:signing/amount-errors]]
+    (let [display-symbol     (wallet.utils/display-symbol token)
+          fee-display-symbol (wallet.utils/display-symbol (tokens/native-currency chain))]
+      [react/view styles/sheet
+       [header sign tx display-symbol fee fee-display-symbol]
+       [separator]
+       (if sign
+         [react/view {:padding-top 20}
+          [password-view sign]]
+         [react/view
+          [contact-item contact]
+          [separator]
+          [token-item token display-symbol]
+          [react/view
+           [list-item/list-item {:type        :small :title (i18n/label :t/send-request-amount)
+                                 :error       amount-error
+                                 :accessories [[acc-text (if amount (str amount) "0")
+                                                (or display-symbol fee-display-symbol)]]}]
+           [separator]]
+          [list-item/list-item
+           {:type        :small :title (i18n/label :t/network-fee) :error gas-error
+            :accessories [[acc-text fee fee-display-symbol] :chevron]
+            :on-press    #(re-frame/dispatch [:signing.ui/open-fee-sheet
+                                              {:content        (fn [] [sheets/fee-bottom-sheet fee-display-symbol])
+                                               :content-height 280}])}]
+          [react/view {:align-items :center :margin-top 16 :margin-bottom 40}
+           [button/primary-button {:on-press  #(re-frame/dispatch [:set :signing/sign {:type :password}])
+                                   :disabled? (or amount-error gas-error)}
+            (i18n/label :t/sign-with-password)]]])])))
+
+(views/defview signing-view [tx window-height]
```

Comment:
use a regular defn here, you don't make any subscription and it will make it more clear what happens

Reply:
(don't forget to deref your atoms)